### PR TITLE
intel/2017b toolchain definition (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

requires ~~https://github.com/hpcugent/easybuild-framework/pull/2215~~ since SHA256 checksums are used

requires ~~#4794~~ for GCC 6.4.0

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
node2159.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/10be848a4bc0d3d8d77779c8bdb1570a for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
node2450.golett.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/24ea6daf0b40ef1bcc31be391e2e23a2 for a full test report.

**Comment by boegel** [comment]
Reply from Intel support on use of Intel compilers 2017 update 4 on top of GCC 7.1.0:

```
In the release note, we specify that we only have support for GCC 6.

https://software.intel.com/en-us/articles/intel-c-compiler-170-for-linux-release-notes-for-intel-parallel-studio-xe-2017

Even the 2018 update will only support GCC 6.3

You might end up having some issues of compatibility.
```

So I guess we'll stick to GCC 6.3.0...

Thoughts @wpoely86, @akesandgren, @damianam, @ocaisa?

**Comment by wpoely86** [comment]
If that is what Intel says, we stick to it? No reason to go all adventurous, especially as GCC 7 has no killer features (or not as far as I know)

**Comment by akesandgren** [comment]
Yeah, better to keep it supported.

Sent with AquaMail for Android
http://www.aqua-mail.com

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 8 (8 easyconfigs in this PR)
node2044.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/c2651063e9df13dfcf601ec0b68cd4a3 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 14 out of 14 (8 easyconfigs in this PR)
node2451.golett.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/e07a4904a840554876c137608f465196 for a full test report.

File: easybuild/easyconfigs/i/intel/intel-2017b.eb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+version = '2017b'
+
+homepage = 'http://software.intel.com/en-us/intel-cluster-toolkit-compiler/'
+description = """Intel Cluster Toolkit Compiler Edition provides Intel C/C++ and Fortran compilers, Intel MPI &
+ Intel MKL."""
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+compver = '2017.4.196'
+gccver = '6.3.0'
```
Comment (boegel) [inline easybuild/easyconfigs/i/intel/intel-2017b.eb]: GCC 6.4.0 is imminent: https://gcc.gnu.org/ml/gcc/2017-06/msg00101.html (hat tip @ocaisa)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
node2092.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/b5fffe046079294b7e58b3b54b5236cf for a full test report.

File: easybuild/easyconfigs/h/HPL/HPL-2.2-intel-2017b.eb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+homepage = 'http://www.netlib.org/benchmark/hpl/'
+description = """HPL is a software package that solves a (random) dense linear system in double precision (64 bits)
+ arithmetic on distributed-memory computers. It can thus be regarded as a portable as well as freely available
+ implementation of the High Performance Computing Linpack Benchmark."""
+
+toolchain = {'name': 'intel', 'version': '2017b'}
+toolchainopts = {'usempi': True}
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = ['http://www.netlib.org/benchmark/%(namelower)s']
```
Comment (verdurin) [inline easybuild/easyconfigs/h/HPL/HPL-2.2-intel-2017b.eb]: Add a checksum here.

File: easybuild/easyconfigs/i/icc/icc-2017.4.196-GCC-6.4.0-2.28.eb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# This is an easyconfig file for EasyBuild, see http://hpcugent.github.io/easybuild
```
Comment (verdurin) [inline easybuild/easyconfigs/i/icc/icc-2017.4.196-GCC-6.4.0-2.28.eb]: Old URL?

File: easybuild/easyconfigs/i/iccifort/iccifort-2017.4.196-GCC-6.4.0-2.28.eb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# This is an easyconfig file for EasyBuild, see http://hpcugent.github.io/easybuild
```
Comment (verdurin) [inline easybuild/easyconfigs/i/iccifort/iccifort-2017.4.196-GCC-6.4.0-2.28.eb]: Old URL again

File: easybuild/easyconfigs/i/ifort/ifort-2017.4.196-GCC-6.4.0-2.28.eb [hunk 5]
```diff
@@ -0,0 +1,36 @@
+# This is an easyconfig file for EasyBuild, see http://hpcugent.github.io/easybuild
```
Comment (verdurin) [inline easybuild/easyconfigs/i/ifort/ifort-2017.4.196-GCC-6.4.0-2.28.eb]: URL

File: easybuild/easyconfigs/i/iimpi/iimpi-2017b.eb [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# This is an easyconfig file for EasyBuild, see http://hpcugent.github.io/easybuild
```
Comment (verdurin) [inline easybuild/easyconfigs/i/iimpi/iimpi-2017b.eb]: URL

File: easybuild/easyconfigs/i/imkl/imkl-2017.3.196-iimpi-2017b.eb [hunk 7]
```diff
@@ -0,0 +1,36 @@
+# This is an easyconfig file for EasyBuild, see http://hpcugent.github.io/easybuild
```
Comment (verdurin) [inline easybuild/easyconfigs/i/imkl/imkl-2017.3.196-iimpi-2017b.eb]: URL again

File: easybuild/easyconfigs/i/impi/impi-2017.3.196-iccifort-2017.4.196-GCC-6.4.0-2.28.eb [hunk 8]
```diff
@@ -0,0 +1,33 @@
+# This is an easyconfig file for EasyBuild, see http://hpcugent.github.io/easybuild
```
Comment (verdurin) [inline easybuild/easyconfigs/i/impi/impi-2017.3.196-iccifort-2017.4.196-GCC-6.4.0-2.28.eb]: URL fix required

File: easybuild/easyconfigs/i/intel/intel-2017b.eb [hunk 9]
```diff
@@ -0,0 +1,25 @@
+easyblock = 'Toolchain'
+
+name = 'intel'
+version = '2017b'
+
+homepage = 'http://software.intel.com/en-us/intel-cluster-toolkit-compiler/'
+description = """Intel Cluster Toolkit Compiler Edition provides Intel C/C++ and Fortran compilers, Intel MPI &
+ Intel MKL."""
```
Comment (verdurin) [inline easybuild/easyconfigs/i/intel/intel-2017b.eb]: Wouldn't it be better to have a proper toolchain description here?

Comment (boegel) [inline easybuild/easyconfigs/i/intel/intel-2017b.eb]: define proper?

Comment (verdurin) [inline easybuild/easyconfigs/i/intel/intel-2017b.eb]: Well, that just reads like some Intel promotional material.

Compare it with the `foss` text.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 14 out of 22 (8 easyconfigs in this PR)
janson.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Core(TM) i7-5600U CPU @ 2.60GHz, Python 2.7.12
See https://gist.github.com/881ce636bef1958aed16188f8f09fff1 for a full test report.

**Comment by boegel** [comment]
@akesandgren You need to make sure a license is available for these...

**Comment by akesandgren** [comment]
Yes i know, i was just a bit too tired this morning. New build is already running.
And stop reading comments i have already deleted :-) :-)

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 19 out of 22 (8 easyconfigs in this PR)
janson.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Core(TM) i7-5600U CPU @ 2.60GHz, Python 2.7.12
See https://gist.github.com/60aea5ef110bc56ed38a00ada12f6e8e for a full test report.

**Comment by boegel** [comment]
@akesandgren Hmm, this one looks trickier, some missing system dependency for Intel MKL? :-/

```
/usr/include/features.h(367): catastrophic error: cannot open source file "sys/cdefs.h"
  #  include <sys/cdefs.h>
```

**Comment by akesandgren** [comment]
No, it's a feature of Debian/Ubuntu that Intel doesn't handle. I have a patch to the icc easyconfig that fixes this. I'll prep a PR for it.

File: easybuild/easyconfigs/i/intel/intel-2017b.eb (hunk 9 above)
Comment (boegel) [reply easybuild/easyconfigs/i/intel/intel-2017b.eb]: Something like this then?

```
Compiler toolchain including Intel compilers, Intel MPI and Intel Math Kernel Library (MKL).
```

Comment (verdurin) [reply easybuild/easyconfigs/i/intel/intel-2017b.eb]: Yes, that's much better.

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 0 out of 10 (8 easyconfigs in this PR)
ca003.camp.thecrick.org - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2640 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/dde40b4e4ce8d53e63de14d7357eca80 for a full test report.

**Comment by verdurin** [comment]
Ignore that failed test...

**Comment by vanzod** [comment]
Test report by @vanzod
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
cermis - Linux debian 8.8, Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz, Python 2.7.9
See https://gist.github.com/9cc4210b65e4103268202e3b2d68c86c for a full test report.

**Comment by vanzod** [comment]
Going in, thanks @boegel!

**Pull request merged**